node-manager/src/nodemanager.rs

```diff
@@ -455,11 +456,13 @@ impl NodeManager {
         amount: u64,
         fee_rate: Option<f32>,
     ) -> Result<String, MutinyJsError> {
-        match self
-            .wallet
-            .send(destination_address, amount, fee_rate)
-            .await
-        {
+        let send_to = Address::from_str(&destination_address)?;
+
+        if send_to.network != self.network {
+            return Err(MutinyJsError::IncorrectNetwork);
+        }
```

Comment:
`sweep_wallet` and `check_address` should probably have this check too, right?

Reply:
Added